shell/common/persistent_cache.cc

```diff
@@ -159,6 +160,27 @@ void PersistentCache::store(const SkData& key, const SkData& data) {
                        std::move(file_name), std::move(mapping));
 }
 
+#if FLUTTER_RUNTIME_MODE != FLUTTER_RUNTIME_MODE_RELEASE
+void PersistentCache::DumpSkp(const SkData& data) {
+  if (is_read_only_ || !IsValid()) {
+    FML_LOG(ERROR) << "Could not dump SKP from read-only or invalid persistent "
+                      "cache.";
+    return;
+  }
+
+  std::stringstream name_stream;
+  int current_index = skp_index_++;
+  skp_index_ %= kMaxSkpIndex;
```

Comment:
Why limit the maximum number of pictures captured? I'd suggest appending the timestamp to the end of the skp as well (`std::to_string(fml::TimePoint::Now()` or something)). Just so its in increasing order. This is a profiling feature anyway so the disk usage is not a concern.